src/AppEntry.tsx

```diff
@@ -33,9 +40,38 @@ if (window.localStorage.getItem("xstate-inspect") !== null) {
 }
 
 const AppWithKeycloak: FunctionComponent = () => {
+  const [ssoProviders, setSSOProviders] = useState<SsoProviderAllOf>();
   console.log("starting appwithkeycloak");
-  const auth = useAuth();
+  let auth = useAuth();
   const dispatch = useDispatch();
+  const getSSOProviders = useSSOProviders();
+
+  useEffect(() => {
+    (async () => {
+      const response = await getSSOProviders();
+      setSSOProviders(response);
+    })();
+  }, [getSSOProviders]);
+
+  const shouldUseMasSSO = (): boolean | undefined => {
+    if (!ssoProviders) return undefined;
+    return ssoProviders?.name === "mas_sso";
+  };
+
+  /**
+   * This is temporarory check.
+   * It will be removed when we will have mas_sso to sso migration fully deployed
+   */
+  if (shouldUseMasSSO() === false) {
```

Comment:
I'm not a fan of all these `undefined` as you know. I'd make this boolean a required info of `AppWithKeycloak` and deal with loading the SSO providers in a parent component of this.